src/custom/pages/Swap/index.tsx

```diff
@@ -124,6 +126,67 @@ const SwapModWrapper = styled(SwapMod)`
   }
 `
 
+const fadeIn = `
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+interface LoadingCowImgProps {
+  maxWidth?: string
+  showLoader: boolean
+}
+
+const LoadingCowImg = styled.img<LoadingCowImgProps>`
+  max-width: ${({ maxWidth = '60px' }) => maxWidth};
+  opacity: ${({ showLoader }) => (showLoader ? 1 : 0)};
+
+  transition: opacity 0.4s ease-in-out;
+`
+
+type LoadingCowProps = {
+  isSoftLoading: boolean
+} & { maxWidth?: string; src: string }
+
+const LOADING_COW_TIMER = 4000
+
+function LoadingCow(props: LoadingCowProps) {
+  const { isSoftLoading, ...rest } = props
+  const [showCow, setShowCow] = useState(isSoftLoading)
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+    // price is being queried
+    if (isSoftLoading) {
+      // isn't currently showing cow or hasnt yet been updated here
+      // so we clear any running timeouts ready to clear local loading state
+      // and essentially reset them
+      clearTimeout(timeout as NodeJS.Timeout)
```

Comment:
Maybe use `timeout && clearTimeout(timeout)` to prevent use of `as`

Reply:
will address in #736